### ISSUE TITLE
ess-insert-S-assign was turned on in SAS 

### ISSUE THREAD

**Issue by ShuguangSun** [emacs-ess/ESS#607]
ess-insert-S-assign is only conditioned on ess-smart-S-assign-key, and is custimized as "_" as default if ess-smart-S-assign-key is non-nil. As they are all global variable and function, so that, ess-insert-S-assign will be binded to "_" for all ess-mode including SAS. 

Could we make ess-smart-S-assign-key as buffer-local variable? Or enable ess-insert-S-assign as hock to R/R inferior mode? I think it is not efficient and necessary to check whether it is in R/R inferior mode in ess-insert-S-assign.

**Comment by jabranham** [comment]
It might make sense to implement `ess-r-mode-map` for stuff like this, what do yall think? @vspinu @lionel-

**Comment by lionel-** [comment]
Seems like a good idea. Especially since we'll have such a map anyway when `ess-r` becomes a proper mode.

**Comment by vspinu** [comment]
Yes. This is a target for 19.04. By then we should have our modes split.
